src/main/drivers/display_ug2864hsweg01.c

```diff
@@ -181,52 +183,75 @@ static bool i2c_OLED_send_cmd(busDevice_t *bus, uint8_t command)
+    for (size_t i = 0 ; i < len ; i++) {
+        if (i2c_OLED_send_cmd(bus, commands[i])) {
+            // XXX Funny, i2cWrite is returning errors!?
+            // return;
+        }
+    }
+
+    return true;
 }
 
-void i2c_OLED_clear_display(busDevice_t *bus)
+static bool i2c_OLED_send_byte(busDevice_t *bus, uint8_t val)
 {
-    i2c_OLED_send_cmd(bus, 0xa6);              // Set Normal Display
-    i2c_OLED_send_cmd(bus, 0xae);              // Display OFF
-    i2c_OLED_send_cmd(bus, 0x20);              // Set Memory Addressing Mode
-    i2c_OLED_send_cmd(bus, 0x00);              // Set Memory Addressing Mode to Horizontal addressing mode
-    i2c_OLED_send_cmd(bus, 0xb0);              // set page address to 0
-    i2c_OLED_send_cmd(bus, 0x40);              // Display start line register to 0
-    i2c_OLED_send_cmd(bus, 0);                 // Set low col address to 0
-    i2c_OLED_send_cmd(bus, 0x10);              // Set high col address to 0
-    for (uint16_t i = 0; i < 1024; i++) {  // fill the display's RAM with graphic... 128*64 pixel picture
-        i2c_OLED_send_byte(bus, 0x00);  // clear
-    }
-    i2c_OLED_send_cmd(bus, 0x81);              // Setup CONTRAST CONTROL, following byte is the contrast Value... always a 2 byte instruction
-    i2c_OLED_send_cmd(bus, 200);               // Here you can set the brightness 1 = dull, 255 is very bright
-    i2c_OLED_send_cmd(bus, 0xaf);              // display on
+    return i2cWrite(bus->i2c.device, bus->i2c.address, 0x40, val);
 }
 
 void i2c_OLED_clear_display_quick(busDevice_t *bus)
 {
-    i2c_OLED_send_cmd(bus, 0xb0);              // set page address to 0
-    i2c_OLED_send_cmd(bus, 0x40);              // Display start line register to 0
-    i2c_OLED_send_cmd(bus, 0);                 // Set low col address to 0
-    i2c_OLED_send_cmd(bus, 0x10);              // Set high col address to 0
+    static const uint8_t i2c_OLED_cmd_clear_display_quick[] = {
+        0xb0, // set page address to 0
+        0x40, // Display start line register to 0
+        0,    // Set low col address to 0
+        0x10, // Set high col address to 0
+    };
+
+    i2c_OLED_send_cmdarray(bus, i2c_OLED_cmd_clear_display_quick, ARRAYLEN(i2c_OLED_cmd_clear_display_quick));
+
     for (uint16_t i = 0; i < 1024; i++) {      // fill the display's RAM with graphic... 128*64 pixel picture
         i2c_OLED_send_byte(bus, 0x00);  // clear
     }
 }
 
+void i2c_OLED_clear_display(busDevice_t *bus)
+{   
+    static const uint8_t i2c_OLED_cmd_clear_display_pre[] = {
+        0xa6, // Set Normal Display
+        0xae, // Display OFF
+        0x20, // Set Memory Addressing Mode
+        0x00, // Set Memory Addressing Mode to Horizontal addressing mode
+    };
+
+    i2c_OLED_send_cmdarray(bus, i2c_OLED_cmd_clear_display_pre, ARRAYLEN(i2c_OLED_cmd_clear_display_pre));
+
+    i2c_OLED_clear_display_quick(bus);
+
+    static const uint8_t i2c_OLED_cmd_clear_display_post[] = {
```

Comment:
Nahh... 2 byte command is not working either...
![saleae_logic_software](https://user-images.githubusercontent.com/14850998/28245752-3f36a4d6-6a48-11e7-9308-07d52cb31ea5.jpg)